python/mxnet/numpy/multiarray.py

```diff
@@ -1993,15 +1993,17 @@ def array(object, dtype=None, ctx=None):
     """
     if ctx is None:
         ctx = current_context()
-    if isinstance(object, ndarray):
+    if isinstance(object, (ndarray, _np.ndarray)):
         dtype = object.dtype if dtype is None else dtype
+    elif isinstance(object, NDArray):
+        raise ValueError("")
     else:
-        dtype = _np.float32 if dtype is None else dtype
-        if not isinstance(object, (ndarray, _np.ndarray)):
-            try:
-                object = _np.array(object, dtype=dtype)
-            except Exception as e:
-                raise TypeError('{}'.format(str(e)))
+        if dtype is None:
+            dtype = object.dtype if hasattr(object, "dtype") else _np.float32
+        try:
+            object = _np.array(object, dtype=dtype)
+        except Exception as e:
+            raise TypeError('{}'.format(str(e)))
```

Comment:
Is such a broad exception handling the right thing? Also, you're stripping all the exception information and just printing the message - nested exception or rethrowing would make more sense here I'd say

Reply:
Also, it's missing the message.

Reply:
It's not missing the message here, the exception is being converted into a string and formatted in the `{}`'s position.

Reply:
You're stripping away the exception stack (in a broad catch all clause) and further information that would give more information. In how far does removing that info add value?

Could you elaborate what you're trying to achieve here? Catching all exceptions and rethrowing them with stripped information is usually not something I see around

Reply:
So here it's simply calling the official numpy's `array` function, but Numpy does not show any "call stack" at all:
```
>>> np.array(1.0, order="MXNET")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: order not understood
```
Catching it here and simply print it out is a more transparent way of showing the users the original error I guess...

Reply:
Okay, thanks for elaborating! Can you add a short comment to document? Fine with me to move forward

Reply:
Let's see if this round of CI check passes, if it does I'll probably prefer to save 1 CI cycle for this... I'll combine the doc into a future PR maybe, lemme know if you're fine with that. BTW error message has been added with respect to your other comment.

Reply:
Fine with me, thanks!